README.md

```diff
@@ -53,18 +53,62 @@ const HelloWorld = () => <h1>Hello, Birst</h1>;
 
 embedInBirst(() => <HelloWorld />);
 ```
+## Local Development and Debugging
+Working on a project within a Birst dashboard itself is a pretty horrendous experience.  I recommend using `webpack-dev-server` with hot reloading for the majority of development and then give it a final test in Birst before shipping!  You can even combine `webpack-dev-server` with `ngrok` to host your local dev content in a Birst dashboard directly.
+
+To test and develop locally just start up `webpack-dev-server` (`Ctrl+Shift+B` if you're using VSCode). This will serve the content to the specified port on local host. By default this is `http://localhost:3000/index.html`.
+
+Navigate to the `index.html` page for the selected component (example: `localhost:3000/src/my-component/my-component.html`). Webpack will automatically rebuild any changes made to the TSX or HTML for that component.
+
+> Note: If you are running locally _without_ ngrok (i.e. not embedded within Birst) then you don't have access to the Birst API. To improve the dev experience, whenever `process.env.NODE_ENV !== 'production'` this library will replace the _real_ BQL API with a mock one that will use `fetch` to resolve the `sampleResponseUrl` for your defined query. It will error with `Unexpected Token at position x` if you try and host locally without a valid `sample.json`.
+
+## Hosting with Ngrok
+To test in Birst you'll need to host the generated code somewhere. Let's assume you have used something like `webpack` to generate a single bundle using dev-server, you can use [ngrok](https://ngrok.com/) to quickly forward the local port that dev-server is running on.
```

Comment:
Question: does this approach mean that hot reloading works within birst?  I think I tried hooking dev server up to ngrok but it didn't quite work correctly...

Reply:
Ah no that was a typo. Related question: I'm not sure if this is intentional but to get dev server to use the BQL I have to edit the `env.js` in the `birst-react-embed` library to `== production`. I've tried removing `--debug` from dev server, I feel like I am missing something 👀

Reply:
I'm pretty sure that `webpack-dev-server` will always set the environment to `"development"` but I might be wrong.

To be honest, I wouldn't waste any time trying to get `webpack-dev-server -> ngrok -> birst dashboard` working.  Better to make sure `webpack-dev-server -> local w/ fake data` works nicely and use `http-server -> ngrok -> birst dashboard` for live testing.

As long as you have the right version of fake data then you should be fine!